src/displayFormatters.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+static const char *getModemPresetDisplayName(meshtastic_Config_LoRaConfig_ModemPreset preset, bool useShortName = false)
+{
+    switch (preset) {
+    case meshtastic_Config_LoRaConfig_ModemPreset_SHORT_SLOW:
+        return useShortName ? "ShortS" : "ShortSlow";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_SHORT_FAST:
+        return useShortName ? "ShortF" : "ShortFast";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_MEDIUM_SLOW:
+        return useShortName ? "MedS" : "MediumSlow";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_MEDIUM_FAST:
+        return useShortName ? "MedF" : "MediumFast";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_LONG_SLOW:
+        return useShortName ? "LongS" : "LongSlow";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_LONG_FAST:
+        return useShortName ? "LongF" : "LongFast";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_LONG_MODERATE:
+        return useShortName ? "LongM" : "LongMod";
+        break;
+    case meshtastic_Config_LoRaConfig_ModemPreset_VERY_LONG_SLOW:
+        return useShortName ? "VeryL" : "VLongSlow";
+        break;
+    default:
+        return useShortName ? "Custom" : "Invalid";
+        break;
+    }
+}
+
+#ifdef ARCH_ESP32
+inline static const char *getWifiDisconnectReasonName(uint8_t reasonCode)
```

Comment:
I would not declare this inline... could be very expensive if someone uses this function elsewhere (e.g. for debug log).

Reply:
We either need to move it to its own file or figure out how to suppress the wUnusedFunction in cppcheck then.  If a consumer like Channels.cpp includes this file for the channel display name method and doesn't use this one, we get warnings.

Reply:
Then put the implementation into displayFormatters.cpp and just do normal function prototypes in displayFormatters.h

Reply:
DisplayFormatters.h:

```
class DisplayFormatters {
public:
    static const char *getModemPresetDisplayName(meshtastic_Config_LoRaConfig_ModemPreset preset, bool useShortName = false);
};
```


DisplayFormatters.cpp:

```
#include "DisplayFormatters.h"

const char * DisplayFormatters::getModemPresetDisplayName(meshtastic_Config_LoRaConfig_ModemPreset preset, bool useShortName = false) {
...
}
```

Screen.cpp (usage):

#include "DisplayFormatters.h"

DisplayFormatters::getModemPresetDisplayName(...);